subsys/net/lib/aws_jobs/src/aws_jobs.c

```diff
@@ -0,0 +1,296 @@
+/*
+ *copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/mqtt.h>
+#include <logging/log.h>
+#include "aws_jobs.h"
+
+LOG_MODULE_REGISTER(aws_jobs, CONFIG_AWS_JOBS_LOG_LEVEL);
+
+
+/**
+ * @brief Local function used to error check snprintf outputs.
+ *
+ * @param[in] ret Return value of snpritnf.
+ * @param[in] size Max size of what would have been the final string.
+ * @param[in] entry Name of the function it was called from.
+ */
+static inline void report_snprintf_err(u32_t ret, u32_t size, char *entry)
+{
+	if (ret >= size) {
+		LOG_ERR("Unable to fit formated string into to allocate"
+			"memory for %s", entry);
+	} else if (ret < 0) {
+		LOG_ERR("Output error for %s was encountered with return value"
+			" %d", entry, ret);
+	}
+}
+
+
+/**
+ * @brief Constructs the get topic for accepted and rejeceted AWS jobs.
+ *
+ * @param[in] client_id Client ID of the current connected MQTT session.
+ * @param[out] topic Outputs the constructed mqtt_topic into the given buffer.
+ */
+static void construct_get_topic(u8_t *client_id, struct mqtt_topic *topic)
+{
+	char get_topic[GET_TOPIC_LEN + 1];
+	int ret = snprintf(get_topic,
+			   GET_TOPIC_LEN,
+			   GET_TOPIC_TEMPLATE,
+			   client_id,
+			   "#");
+	report_snprintf_err(ret, GET_TOPIC_LEN, "get_topic");
+
+	topic->topic.utf8 = get_topic;
+	topic->topic.size = ret + 1;
+	topic->qos = MQTT_QOS_1_AT_LEAST_ONCE;
+	return;
```

Comment:
And at this point, `get_topic` is gone. We can't use a stack variable for the buffer, at least not within this helper function.

I see that this issue happens in other places as well, it should all be fixed.